hazelcast-client/src/test/java/com/hazelcast/client/heartbeat/ClientHeartbeatTest.java

```diff
@@ -102,6 +102,39 @@ public void connectionRemoved(Connection connection) {
         assertOpenEventually(countDownLatch);
     }
 
+    @Test
+    public void testNonOwnerConnectionClosed_whenHeartbeatStopped() throws InterruptedException {
+        hazelcastFactory.newHazelcastInstance();
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(getClientConfig());
+        HazelcastInstance instance = hazelcastFactory.newHazelcastInstance();
+
+        HazelcastClientInstanceImpl clientImpl = getHazelcastClientInstanceImpl(client);
+        final ClientConnectionManager connectionManager = clientImpl.getConnectionManager();
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                assertEquals(2, connectionManager.getActiveConnections().size());
+            }
+        });
+
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+        connectionManager.addConnectionListener(new ConnectionListener() {
+            @Override
+            public void connectionAdded(Connection connection) {
+
+            }
+
+            @Override
+            public void connectionRemoved(Connection connection) {
+                countDownLatch.countDown();
```

Comment:
I think you should check that this is non-owner connection, don't you?

Reply:
I did not see the need because we know that I am blocking messages to non-owner member. But I don't see any harm adding one more check,so I will add it.